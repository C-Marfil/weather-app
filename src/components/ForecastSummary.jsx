import React from "react";
import PropTypes from "prop-types";

function ForecastSummary(props) {
  const { date, description, icon, temperature } = props;
  return (
    <div className="forecast-summary">
      <div className="forecast-summary__date">{date}</div>
      <div className="forecast-summary__temperature">{temperature.max}</div>
      <div className="forecast-summary__description">{description}</div>
      <div className="forecast-summary__icon">{icon}</div>
    </div>
  );
}

ForecastSummary.propTypes = {
  date: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  temperature: PropTypes.shape({
    min: PropTypes.number,
    max: PropTypes.number,
  }).isRequired,
};

export default ForecastSummary;
