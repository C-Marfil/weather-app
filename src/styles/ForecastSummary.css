.forecast-summary {
  background-color: rgb(70, 56, 56);
  color: #fff;
  opacity: 0.5;
  border-radius: 10px;
  padding: 10px;
  display: flex;
  flex-direction: column;
}
