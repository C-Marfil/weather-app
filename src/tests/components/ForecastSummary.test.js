import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../../components/ForecastSummary";

describe("ForecastSummary", () => {
  const validProps = {
    date: 1111111,
    description: "Stub description",
    icon: "stubIcon",
    temperature: {
      min: 12,
      max: 12,
    },
  };

  it("renders the correct forecast for the day", () => {
    const { asFragment } = render(
      <ForecastSummary
        date={validProps.date}
        description={validProps.description}
        icon={validProps.icon}
        temperature={validProps.temperature}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
