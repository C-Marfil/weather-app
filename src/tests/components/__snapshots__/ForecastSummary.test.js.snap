// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders the correct forecast for the day 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
  >
    <div
      class="forecast-summary__date"
    >
      1111111
    </div>
    <div
      class="forecast-summary__temperature"
    >
      12
    </div>
    <div
      class="forecast-summary__description"
    >
      Stub description
    </div>
    <div
      class="forecast-summary__icon"
    >
      stubIcon
    </div>
  </div>
</DocumentFragment>
`;
